internal/charmstore/store.go

```diff
@@ -804,6 +819,25 @@ func (s *Store) Publish(url *router.ResolvedURL, channels ...params.Channel) err
 	return nil
 }
 
+func (s *Store) checkPublishedResources(entity *mongodoc.Entity, resources map[string]int) error {
+	knownResources, _, err := s.charmResources(entity.BaseURL)
```

Comment:
We can't do that because the charm might require us to have resources (that check is implemented in the next PR).